src/libponyrt/mem/heap.c

```diff
@@ -524,7 +525,12 @@ void ponyint_heap_used(heap_t* heap, size_t size)
 
 bool ponyint_heap_startgc(heap_t* heap)
 {
-  if(heap->used <= heap->next_gc)
+  // don't run GC if there's no heap used
+  if(heap->used == 0)
+    return false;
+
+  // don't run GC if haven't reached next threshold and aren't under memory pressure
+  if((heap->used <= heap->next_gc) && !ponyint_pool_mem_pressure())
```

Comment:
what if instead of tackling in this fashion, we capped next_gc?

this might be a bad idea as it is overloading the meaning of next_gc otoh, it literally is what we are doing here.

Reply:
I'm not sure I follow. How exactly would we cap `next_gc` in relation to what `ponyint_pool_mem_pressure()` checks for?

Reply:
instead of using "pool_mem_pressure" when setting a new `next_gc` value, don't allow it to be set to more than the value which currently triggers true from `ponyint_pool_mem_pressure`.

Reply:
Ah, okey. Thanks for clarifying. I'm not sure that accomplishes the same thing. It is very likely some actors will get their `next_gc` set prior to other actors allocating more memory that the `next_gc` would be above the value that triggers true from `ponyint_pool_mem_pressure` for the some actors.

Reply:
you are right. my idea makes no sense. i blame the migraine.

Reply:
migraine no good. 8*/